keras_cv/metrics/object_detection/box_count_delta.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+import tensorflow.keras as keras
+import tensorflow.keras.initializers as initializers
+
+from keras_cv import bounding_box
+
+
+class BoxCountDelta(keras.metrics.Metric):
+    """BoxCountDelta counts the difference of counts of predicted and true boxes.
+
+    BoxCountDelta looks at the number of boxes in the ground truth dataset
+    and counts the delta between that number and the number of boxes your object
+    detection model predicted for that image. This is primarily useful when
+    attempting to tune the confidence threshold of your
+    `MultiClassNonMaxSuppression` layer in an object detection model.  If this
+    metric is high, it indicates that your model is making too many or too few
+    predictions.  Ideally this metric will be zero.
+
+    Args:
+      mode: (Optional) either 'relative' or 'absolute'.  When set to 'absolute',
```

Comment:
Let's mention what relative does, and when that might be useful.

Reply:
I've added a paragraph on how to interpret this.